src/Serializers/StatementListSerializer.php

```diff
@@ -69,12 +68,14 @@ public function serialize( $object ) {
 	private function getSerialized( StatementList $statementList ) {
 		$serialization = array();
 
-		$byPropertyIdGrouper = new ByPropertyIdGrouper( $statementList );
+		foreach ( $statementList->toArray() as $statement ) {
+			$idSerialization = $statement->getPropertyId()->getSerialization();
 
-		foreach ( $byPropertyIdGrouper->getPropertyIds() as $propertyId ) {
-			$serialization[$propertyId->getSerialization()] = $this->getSerializedStatements(
-				$byPropertyIdGrouper->getByPropertyId( $propertyId )
-			);
+			if ( !array_key_exists( $idSerialization, $serialization ) ) {
+				$serialization[$idSerialization] = array();
+			}
```

Comment:
It works without this if block... not sure if that is good style though, or if it will continue to work in later PHP versions